src/visibility.js

```diff
@@ -0,0 +1,173 @@
+// @flow
+import {BasePlugin, Utils} from 'playkit-js';
+import './style.css';
+import {DismissibleFloatingButtonComponent} from './components/dismissible/dismissible';
+import 'intersection-observer';
+
+const FLOATING_ACTIVE_CLASS: string = 'playkit-floating-active';
+const FLOATING_CONTAINER_CLASS: string = 'playkit-floating-container';
+const FLOATING_POSTER_CLASS: string = 'playkit-floating-poster';
+
+/**
+ * Visibility class.
+ * @classdesc
+ */
+class Visibility extends BasePlugin {
+  appTargetContainer: HTMLElement | null;
+  floatingContainer: HTMLElement;
+  floatingPoster: HTMLElement;
+  observer: window.IntersectionObserver;
+  everStartedPlaying: boolean = false;
+  dismissed: boolean = false;
+
+  /**
+   * The default configuration of the plugin.
+   * @type {Object}
+   * @static
+   */
+  static defaultConfig: Object = {
+    threshold: 0.5
+  };
+
+  getUIComponents() {
+    return [
+      {
+        label: 'dismissibleFloatingButtonComponent',
+        presets: ['Playback', 'Live', 'Error', 'Ads', 'Idle'],
+        container: 'TopBarRightControls',
+        get: DismissibleFloatingButtonComponent,
+        props: {
+          onClose: () => {
+            this._dismissed();
+          }
+        }
+      }
+    ];
+  }
+
+  /**
+   * @static
+   * @public
+   * @returns {boolean} - Whether the plugin is valid.
+   */
+  static isValid(): boolean {
+    return true;
+  }
+
+  /**
+   * @constructor
+   * @param {string} name - The plugin name.
+   * @param {Player} player - The player instance.
+   * @param {Object} config - The plugin config.
+   */
+  constructor(name: string, player: Player, config: Object) {
+    super(name, player, config);
+    this.reset();
```

Comment:
It could be in init on `loadMedia` instead? cause all of it will be relevant after loadMedia I guess and won't after reset, am I right?

Reply:
It should be initiated only once when player is loaded and not every change media

Reply:
But actually maybe it is better is stop observing in reset and observing again in load media

Reply:
I thought about it again. The floating player shouldn't be effected upon reset / loadMedia.
For example, when we are in between videos of a playlist. The plugin should be initiated in the constructor and destroyed in the destory. Thats it